test/fake-intake/fakeintake/fakeintake.go

```diff
@@ -66,71 +68,89 @@ type postPayloadResponse struct {
 	Errors []string `json:"errors"`
 }
 
-func (fi *FakeIntake) postPayload(w http.ResponseWriter, req *http.Request) {
-	fmt.Println("Handling PostPayload request")
+func (fi *FakeIntake) handleDatadogRequest(w http.ResponseWriter, req *http.Request) {
 	if req == nil {
-		writePostPayloadResponse(w, []string{"invalid request, nil request"})
+		response := buildPostResponse(errors.New("invalid request, nil request"))
+		writeHttpResponse(w, response)
 		return
 	}
 
+	log.Printf("Handling Datadog %s request to %s", req.Method, req.URL.Path)
+
+	if req.Method == http.MethodGet {
+		writeHttpResponse(w, httpResponse{
+			statusCode: http.StatusOK,
+		})
+		return
+	}
+
+	// from now on accept only POST requests
 	if req.Method != http.MethodPost {
-		writePostPayloadResponse(w, []string{fmt.Sprintf("invalid request with route %s and method %s", req.URL.Path, req.Method)})
+		response := buildPostResponse(fmt.Errorf("invalid request with route %s and method %s", req.URL.Path, req.Method))
+		writeHttpResponse(w, response)
```

Comment:
You can return `405 Method Not Allowed`.

Reply:
This will return `400  - Bad Request`, can improve it to be more explicit, but `400` should be enough